contractcourt/htlc_success_resolver.go

```diff
@@ -116,139 +114,71 @@ func (h *htlcSuccessResolver) ResolverKey() []byte {
 // anymore. Every HTLC has already passed through the incoming contest resolver
 // and in there the invoice was already marked as settled.
 //
-// TODO(roasbeef): create multi to batch
-//
 // NOTE: Part of the ContractResolver interface.
+//
+// TODO(yy): refactor the interface method to return an error only.
 func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
-	// If we're already resolved, then we can exit early.
-	if h.resolved {
-		return nil, nil
-	}
-
-	// If we don't have a success transaction, then this means that this is
-	// an output on the remote party's commitment transaction.
-	if h.isRemoteCommitOutput() {
-		return h.resolveRemoteCommitOutput()
-	}
-
-	// Otherwise this an output on our own commitment, and we must start by
-	// broadcasting the second-level success transaction.
-	secondLevelOutpoint, err := h.broadcastSuccessTx()
-	if err != nil {
-		return nil, err
-	}
-
-	// To wrap this up, we'll wait until the second-level transaction has
-	// been spent, then fully resolve the contract.
-	return nil, h.resolveSuccessTxOutput(*secondLevelOutpoint)
-}
-
-// broadcastSuccessTx handles an HTLC output on our local commitment by
-// broadcasting the second-level success transaction. It returns the ultimate
-// outpoint of the second-level tx, that we must wait to be spent for the
-// resolver to be fully resolved.
-func (h *htlcSuccessResolver) broadcastSuccessTx() (
-	*wire.OutPoint, error) {
-
-	// If we have non-nil SignDetails, this means that have a 2nd level
-	// HTLC transaction that is signed using sighash SINGLE|ANYONECANPAY
-	// (the case for anchor type channels). In this case we can re-sign it
-	// and attach fees at will. We let the sweeper handle this job.  We use
-	// the checkpointed outputIncubating field to determine if we already
-	// swept the HTLC output into the second level transaction.
-	if h.isZeroFeeOutput() {
-		return h.broadcastReSignedSuccessTx()
-	}
-
-	// Otherwise we'll publish the second-level transaction directly and
-	// offer the resolution to the nursery to handle.
-	log.Infof("%T(%x): broadcasting second-layer transition tx: %v",
-		h, h.htlc.RHash[:], spew.Sdump(h.htlcResolution.SignedSuccessTx))
+	var err error
 
-	// We'll now broadcast the second layer transaction so we can kick off
-	// the claiming process.
-	err := h.resolveLegacySuccessTx()
-	if err != nil {
-		return nil, err
+	// If the HTLC has custom records, then for now we'll pause resolution.
+	//
+	// TODO(roasbeef): Implement resolving HTLCs with custom records
```

Comment:
Should be removed once rebased on master.

Reply:
done!